crates/cairo-profiler/src/trace_reader/sample.rs

```diff
@@ -28,13 +28,15 @@ impl FunctionCall {
 pub enum InternalFunctionCall {
     Inlined(FunctionName),
     NonInlined(FunctionName),
+    Syscall(FunctionName),
```

Comment:
I know this messes up the abstraction layers a bit, but I wonder if it wouldn't be better from a safety point of view to have `Syscall(DeprecatedSyscallSelector)` so you don't have to convert back and forth between string and this type when mapping syscall calls to resources. Would allow you to get rid of `impl FromStr for DeprecatedSyscallSelector`. 

In the future we can implement an intermediate structure for this to make sure the abstraction doesn't leak to `profile_builder` if it becomes a problem.

Reply:
It's an interesting idea, but I'm afraid it would break many things at this point. Let me create an issue and think about ramifications to this. Is that ok?

Maybe it could be done 'together' with splitting the DeprecartedSyscallSelector into syscall and libfuncs enums?

Reply:
https://github.com/software-mansion/cairo-profiler/issues/117 :+1: